src/provider/view/utils/Grouping.ts

```diff
@@ -66,12 +66,9 @@ export function getGroupTitle(notification: StoredNotification, groupingType: Gr
  * @param groupMethod Grouping type to use for sorting.
  */
 export function groupNotifications(notifications: StoredNotification[], groupMethod: GroupingType): Map<string, Group> {
-    // TODO: This sorting should be removed and changed to use insert-sort inside the reduce below
-    // Pre-sort notifications by date (groups will then also be sorted by date)
-    notifications.sort((a: StoredNotification, b: StoredNotification) =>
-        b.notification.date.valueOf() - a.notification.date.valueOf());
     // Reduce the notifications array into a Map of notifications grouped by their title
-    const groupMap = notifications.reduce((groups: Map<string, Group>, currentNotification: StoredNotification) => {
+    // Notifications are already in chronological order so we can just reduce from the right
```

Comment:
Are they always in chronological order?..

We don't explicitly order them ever, so I think they'll be in "creation" order. But since apps can provide a date, they're not necessarily in date order.

Reply:
Ah yeah, I forgot about that. Can we perform sorting when we reduce instead?

Reply:
We could, but would have to write the 'insertion sort' code by hand. I think a `sort()` followed by `reduce()` would make most sense.

Reply:
Reverted back to sorting.